test/fixtures/detect-existing-browser-api-object/content.js

```diff
@@ -10,12 +10,27 @@ test("browser api object in content script", (t) => {
     // On Firefox, window is not the global object for content scripts, and so we expect window.browser to not
     // be defined.
     t.equal(window.browser, undefined, "window.browser is expected to be undefined on Firefox");
+
+    try {
+      await browser.storage.local.set({a: 'a'.repeat(10000000)});
+      t.fail('It should throw when set an object over quota');
+    } catch (error) {
+      t.ok(error instanceof Error);
+    }
```

Comment:
The test did actually run in the CI job and this is the failure logged in the CI job:

```
browser api object in content script (Firefox)
    ✔  a global browser API object should be defined
    ✔  a global chrome API object should be defined
    ✔  browser API object should not be changed on Firefox
    ✔  window.browser is expected to be undefined on Firefox
    ✖  It should throw when set an object over quota
    -------------------------------------------------
        operator: deepEqual
        diff: 
        source: assert@moz-extension://40d035e9-8992-4e15-a880-d7950b4594e9/tape.js:8480:54
        bound@moz-extension://40d035e9-8992-4e15-a880-d7950b4594e9/tape.js:8332:32
        [72]</</</Test.prototype.fail@moz-extension://40d035e9-8992-4e15-a880-d7950b4594e9/tape.js:8574:10
        bound@moz-extension://40d035e9-8992-4e15-a880-d7950b4594e9/tape.js:8332:32
        @moz-extension://40d035e9-8992-4e15-a880-d7950b4594e9/content.js:16:9
```

It is failing because on Firefox we don't enforce the quota for storage.local in the same way that Chrome does, but we do enforce the quota (and same limits) for the storage.sync API and so changing this to use `chrome.storage.sync.set` should make the test to pass also on Firefox.

I did run it locally as well using the command suggested in the doc page I linked before:
```
TEST_BROWSER_TYPE=firefox npm run test-integration | ./node_modules/.bin/tap-nirvana
```

isn't that working for you? any useful output to understand why?